### PR TITLE
frontend: add storybook

### LINKED ISSUE #24: ui: add Storybook
**Description**

Add [Storybook](https://github.com/storybookjs/storybook) to Clutch to provide a design system for components and workflows.

Complexity [S/M/L]: M

### PR DESCRIPTION
**Pull request by dschaller**
### Description

Initial storybook implementation.

**Note**: Due to the babel changes you MUST remove your root node_modules directory before starting storybook for the first time.

In subsequent PRs we can:
- build out the component library
- add a friendlier landing page
- add netlify support to host and publish storybook

Screen Shot 2020-09-29 at 1 16 03 PM

### Testing Performed

manual

### GitHub Issue

Fixes #24

### CODE REVIEW

File: frontend/.storybook/theme.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { create } from '@storybook/theming/create';
```
Comment (mcutalo88) [inline frontend/.storybook/theme.js:1]: I assume this theme is to theme storybook its self, would be cool if we could use the clutch theme one day.

File: frontend/packages/core/src/AppLayout/stories/logo.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
```
Comment (mcutalo88) [inline frontend/packages/core/src/AppLayout/stories/logo.stories.tsx:1]: Just for my understanding, it seems like you need to have a `stories` dir in each component and thats automatically pulled in by storybook at build time?

File: frontend/.storybook/theme.js (hunk 1 above)
Comment (dschaller) [reply frontend/.storybook/theme.js:1]: Yeah that would be ideal. I think the best we can get is reusing the same palette sine the keys don't map.

File: frontend/packages/core/src/AppLayout/stories/logo.stories.tsx (hunk 2 above)
Comment (dschaller) [reply frontend/packages/core/src/AppLayout/stories/logo.stories.tsx:1]: yupp, we can update this to be any pattern: https://github.com/lyft/clutch/pull/489/files#diff-c0daeb9d52f8778948a4e8a2835a953dR3

File: frontend/.storybook/preview.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { Theme } from "./../packages/core/src/AppProvider/themes";
+
+export const parameters = {
+  actions: { argTypesRegex: "^on[A-Z].*" },
```
Comment (danielhochman) [inline frontend/.storybook/preview.js:5]: comment what the intent is here

**Review by danielhochman — COMMENTED** [review]
awesome!

**Review by mcutalo88 — COMMENTED** [review]
Just ran this locally and it looks great!

File: frontend/.storybook/preview.js (hunk 3 above)
Comment (dschaller) [reply frontend/.storybook/preview.js:5]: I'll actually remove this for now since we aren't really using it

**Review by danielhochman — APPROVED** [review]

**Pull request merged**